app/components/post_attachment_opengraph/post_attachment_opengraph.js

```diff
@@ -370,3 +318,55 @@ const getStyleSheet = makeStyleSheetFromTheme((theme) => {
         },
     };
 });
+
+const getBestImageUrlAndDimensions = (props) => {
```

Comment:
Why move it outside the component? and if this is the case then do move it to an utility file and pass the appropriate arguments `(opeGraphData, imagesMetadata)`

Reply:
Well, it was actually all moved out due to getDerivedStateFromProps, but then we eliminated that.  Now we eliminated that, we could move it back in.

Good catch.  Part of clean-up of clean-up of clean-up. :)